.github/pkl-workflows/ci.pkl

```diff
@@ -0,0 +1,66 @@
+amends "package://pkg.pkl-lang.org/github.com/stefma/pkl-gha/com.github.action@0.0.3#/GitHubAction.pkl"
+import "modules/Steps.pkl"
+
+name = "Continuous Integration"
+
+on {
+  pull_request {}
+  push {
+    branches {
+      "main"
+    }
+  }
+}
+
+permissions {
+  contents = "read"
+}
+
+jobs {
+  ["test-typescript"] {
+    name = "TypeScript Tests"
+    `runs-on` = "ubuntu-latest"
+    steps {
+      ...Steps.checkoutAndSetupNode
+      new {
+        name = "Check Format"
+        run = "npm run format:check"
+      }
+      new {
+        name = "Lint"
+        run = "npm run lint"
+      }
+      new {
+        name = "Test"
+        run = "npm run ci-test"
+      }
+    }
+  }
+  ["test-action"] {
+    strategy {
+      matrix {
+        ["os"] = new {
```

Comment:
If this has been typed correctly in your base module, you shouldn't need `= new {}` to create a new list, the override syntax `["os"] { ... }` should work. The only reason to do that typically with a Listing would be if there are default values in the parent object and you want to "start fresh" with a new Listing. Let me go check the typing in your parent module.

Reply:
Looking at https://github.com/StefMa/pkl-gha/blob/main/GitHubAction.pkl#L592, I think there's extra complexity because `matrix` is typed as `Listing<String|*Dynamic>`. Can it really be a Dynamic object?

Reply:
Gotcha - to me that looks more like it could be `Mapping<String, String>` - remember that `Dynamic` can contain properties, entries, AND elements, eg:
```
new {
  "an entry"
  prop1 = "value"
  ["entry one"] = "value2"
}
```

Note that this is actually not representable in JSON or YAML at all (below is from the REPL):

```
pkl> x = new {
>   "an entry"
>   prop1 = "value"
>   ["entry one"] = "value2"
> }
pkl> (new JsonRenderer {}).renderDocument(x)
–– Pkl Error ––
Cannot render object with both elements and properties/entries as JSON.
Object: new Dynamic { prop1 = "value"; ["entry one"] = "value2"; "an entry" }
```

But that's an aside, not important to this PR. Just info for you and your library

Reply:
Thanks for the hint!
Do you now how I could model this better/differently? 🤔